source/fundamentals/data-formats/documents.txt

```diff
@@ -284,6 +289,96 @@ For more information on retrieving and manipulating MongoDB data, see our
 For more information on this class, see the
 :java-docs:`BsonDocument API documentation </apidocs/bson/org/bson/BsonDocument.html>`.
 
+JsonObject
+-------------
+
+The ``JsonObject`` class acts as a wrapper for JSON strings.
+If you want to work primarily with JSON strings in your application, 
+use the ``JsonObject`` class. For applications that work with JSON
+strings, ``JsonObject`` is a more performant alternative to both ``Document``
+and ``BsonDocument`` as ``JsonObject`` does not create an intermediate
+``Map`` object.
+
+By default, ``JsonObject`` stores :manual:`Extended JSON </reference/mongodb-extended-json/>`.
+You can customize the format of JSON in ``JsonObject`` by specyfying a
+:java-docs:`JsonObjectCodec</apidocs/bson/org/bson/codecs/JsonObjectCodec.html>`
+and passing it a
+:java-docs:`JsonWriterSettings</apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+object. For more information on JSON formats, see
+our :doc:`Extended JSON guide </fundamentals/data-formats/document-data-format-extended-json/>`.
+For more information on specifying codecs, see our 
+:doc:`Codecs guide </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate a sample ``JsonObject`` 
+instance wrapping an Extended JSON string containing different types of key value pairs:
+
+.. code-block:: java
+
+   String ejsonStr =  "{"
+                     +" \"_id\": {"
+                     +"  \"$oid\": \"6035210f35bd203721c3eab8\""
+                     +" },"
+                     +" \"name\": \"Gabriel Garc\\u00eda M\\u00e1rquez\","
+                     +" \"dateOfDeath\": {"
+                     +"  \"$date\": \"2014-04-17T04:00:00Z\""
+                     +" },"
+                     +" \"novels\": ["
+                     +"  {"
+                     +"   \"title\": \"One Hundred Years of Solitude\","
+                     +"   \"yearPublished\": 1967"
+                     +"  },"
+                     +"  {"
+                     +"   \"title\": \"Chronicle of a Death Foretold\","
+                     +"   \"yearPublished\": 1981"
+                     +"  },"
+                     +"  {"
+                     +"   \"title\": \"Love in the Time of Cholera\","
+                     +"   \"yearPublished\": 1985"
+                     +"  }"
+                     +" ]"
+                     +"}";
+   JsonObject author = new JsonObject(ejsonStr);
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection()</apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``JsonObject`` class. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
```

Comment:
nitpick: ordering

Maybe we can break this up into steps and follow each step with the code that does the thing. Also, can the first sentence be rewritten to avoid using the same word so close together?

Reply:
Thanks for the feedback @terakilobyte. This language and code structure are inherited from the other examples on the page. For example: 

> To insert this document into a collection, instantiate a collection using getCollection(), specifying the BasicDBObject class. Then, call the insertOne operation as follows:

I think restructuring the examples in the Documents section is a separate unit of work from adding the `JsonObject` documentation. I think grouping this and https://github.com/mongodb/docs-java/pull/57#discussion_r581353216 into a separate ticket makes sense.  If you disagree, I can restructure the three sections as part of this PR however.